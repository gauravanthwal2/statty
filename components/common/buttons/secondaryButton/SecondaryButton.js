import styles from "./SecondaryButton.module.css";

const SecondaryButton = ({ text, ...buttoProps }) => {
  return (
    <button
      className={styles.secondaryButton}
      {...buttoProps}
      data-testid="secondary"
    >
      {text}
    </button>
  );
};

export default SecondaryButton;
