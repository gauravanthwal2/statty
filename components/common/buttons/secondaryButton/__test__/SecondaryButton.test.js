import userEvent from "@testing-library/user-event";
import {
  renderWithProvider,
  cleanup,
  fireEvent,
} from "../../../../../utils/test-utils";
import SecondaryButton from "../SecondaryButton";

const AddParticipantButtonText = "Add Participant";
const UpdateParticipantButtonText = "Update Participant";

describe("button renders", () => {
  afterEach(() => cleanup());
  test("testing primary button on add participant", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <SecondaryButton text={AddParticipantButtonText} onClick={handleSubmit} />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(AddParticipantButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("testing primary button on update participant", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <SecondaryButton
        text={UpdateParticipantButtonText}
        onClick={handleSubmit}
      />
    );
    const button = getByRole("button");
    expect(button).toHaveTextContent(UpdateParticipantButtonText);
    fireEvent.click(button);
    expect(handleSubmit).toHaveBeenCalledTimes(1);
  });

  test("check if button has a css class", () => {
    const handleSubmit = jest.fn();
    const { getByRole } = renderWithProvider(
      <SecondaryButton
        text={UpdateParticipantButtonText}
        onClick={handleSubmit}
      />
    );
    const button = getByRole("button");
    expect(button).toHaveClass("secondaryButton");
  });

  test("renders secondary button  unchanged", () => {
    const handleSubmit = jest.fn();
    const { container } = renderWithProvider(
      <SecondaryButton text={AddParticipantButtonText} onClick={handleSubmit} />
    );
    expect(container).toMatchSnapshot();
  });

  test("check  button attributes", () => {
    const handleSubmit = jest.fn();
    const props = {
      type: "submit",
    };
    const { getByRole } = renderWithProvider(
      <SecondaryButton
        text={UpdateParticipantButtonText}
        onClick={handleSubmit}
        {...props}
      />
    );
    const button = getByRole("button");
    expect(button).not.toHaveAttribute("disabled");
    expect(button).toHaveAttribute("type", "submit");
    expect(button).not.toHaveAttribute("type", "button");
  });
});
