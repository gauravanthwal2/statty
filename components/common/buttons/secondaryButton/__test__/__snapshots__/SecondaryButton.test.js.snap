// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders renders secondary button  unchanged 1`] = `
<div>
  <button
    class="secondaryButton"
    data-testid="secondary"
  >
    Add Participant
  </button>
</div>
`;
